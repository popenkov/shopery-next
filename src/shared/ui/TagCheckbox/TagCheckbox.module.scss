@import '@/shared/styles/_breakpoints';

.tag {
  cursor: pointer;
}

.tagText {
  display: block;
  padding: 6px 16px;
  color: var(--monochrome-900);
  background-color: var(--monochrome-50);
  border-radius: 30px;
}

.tagInput {
  position: absolute;
  width: 0;
  height: 0;
  cursor: pointer;
  opacity: 0;

  &:checked ~ .tagText {
    color: var(--front-primary-inversed);
    background-color: var(--primary);
  }
}
