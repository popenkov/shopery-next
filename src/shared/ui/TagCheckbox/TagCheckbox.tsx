import { ChangeEvent, DetailedHTMLProps, FC, HTMLAttributes } from 'react';

import { Text } from '../Text';

import cls from './TagCheckbox.module.scss';


































































































interface Props
    extends Omit<
        DetailedHTMLProps<HTMLAttributes<HTMLLabelElement>, HTMLLabelElement>,
        'onChange'
    > {
    value: string;
    name: string;
    title: string;
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

export const TagCheckbox: FC<Props> = ({ value, name, title, onChange }) => {
    return (
        <label className={cls.tag}>
            <input
                className={cls.tagInput}
                type="checkbox"
                name={name}
                value={value}
                onChange={onChange}
            />
            <Text variant="body_s" className={cls.tagText}>
                {title}
            </Text>
        </label>
    );
};
