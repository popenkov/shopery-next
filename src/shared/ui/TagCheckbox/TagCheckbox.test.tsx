import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { TagCheckbox } from './TagCheckbox';

describe('TagCheckbox component', () => {
  it('renders correctly', () => {
    const onChange = jest.fn();
    const { container } = render(
      <TagCheckbox value="value" name="name" title="Title" onChange={onChange} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correct class names', () => {
    const { container } = render(
      <TagCheckbox value="value" name="name" title="Title" onChange={() => {}} />,
    );
    expect(container.querySelector('.tag')).toBeInTheDocument();
    expect(container.querySelector('.tagInput')).toBeInTheDocument();
    expect(container.querySelector('.tagText')).toBeInTheDocument();
  });

  it('renders correct text', () => {
    const { getByText } = render(
      <TagCheckbox value="value" name="name" title="Title" onChange={() => {}} />,
    );
    expect(getByText('Title')).toBeInTheDocument();
  });

  it('calls onChange when input changes', () => {
    const onChange = jest.fn();
    const { getByRole } = render(
      <TagCheckbox value="value" name="name" title="Title" onChange={onChange} />,
    );
    const input = getByRole('checkbox');
    fireEvent.click(input);
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('has correct input attributes', () => {
    const { getByRole } = render(
      <TagCheckbox value="value" name="name" title="Title" onChange={() => {}} />,
    );
    const input = getByRole('checkbox');
    expect(input.getAttribute('type')).toBe('checkbox');
    expect(input.getAttribute('name')).toBe('name');
    expect(input.getAttribute('value')).toBe('value');
  });
});
