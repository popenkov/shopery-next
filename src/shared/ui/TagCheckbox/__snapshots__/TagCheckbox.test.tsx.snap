// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCheckbox component renders correctly 1`] = `
<div>
  <label
    class="tag"
  >
    <input
      class="tagInput"
      name="name"
      type="checkbox"
      value="value"
    />
    <p
      class="text tagText primary body_s left regular"
    >
      Title
    </p>
  </label>
</div>
`;
